Straight_skeleton_2/include/CGAL/Straight_skeleton_builder_traits_2.h

```diff
@@ -494,33 +680,239 @@ class Straight_skeleton_builder_traits_2_impl<Tag_true,K> : public Straight_skel
+  }
+
+  Oriented_side_of_event_point_wrt_bisector_2
+  get(Oriented_side_of_event_point_wrt_bisector_2 const* = 0 ) const
+  {
+    return Oriented_side_of_event_point_wrt_bisector_2( typename Exact::Oriented_side_of_event_point_wrt_bisector_2(
+                                                          mExact_traits.mCoeff_cache),
+                                                        typename Filtering::Oriented_side_of_event_point_wrt_bisector_2(
+                                                          mApproximate_traits.mCoeff_cache) );
+  }
+
+  Do_ss_event_exist_2
+  get(Do_ss_event_exist_2 const* = 0 ) const
+  {
+    return Do_ss_event_exist_2( typename Exact::Do_ss_event_exist_2(
+                                  mExact_traits.mTime_cache, mExact_traits.mCoeff_cache),
+                                typename Filtering::Do_ss_event_exist_2(
+                                  mApproximate_traits.mTime_cache, mApproximate_traits.mCoeff_cache) );
+  }
+
+  Are_ss_events_simultaneous_2
+  get(Are_ss_events_simultaneous_2 const* = 0 ) const
+  {
+    return Are_ss_events_simultaneous_2( typename Exact::Are_ss_events_simultaneous_2(
+                                           mExact_traits.mTime_cache, mExact_traits.mCoeff_cache),
+                                         typename Filtering::Are_ss_events_simultaneous_2(
+                                           mApproximate_traits.mTime_cache, mApproximate_traits.mCoeff_cache) );
+  }
+
+  Construct_ss_event_time_and_point_2
+  get(Construct_ss_event_time_and_point_2 const* = 0 ) const
+  {
+    return Construct_ss_event_time_and_point_2( typename Exact::Construct_ss_event_time_and_point_2(
+                                                  mExact_traits.mTime_cache, mExact_traits.mCoeff_cache),
+                                                typename Filtering::Construct_ss_event_time_and_point_2(
+                                                  mApproximate_traits.mTime_cache, mApproximate_traits.mCoeff_cache) );
+  }
+
+// constructor of trisegments using global id stored in the traits
+  Construct_ss_trisegment_2
+  get( Construct_ss_trisegment_2 const* = 0 ) const
+  {
+    return Construct_ss_trisegment_2(*this);
+  }
+
+// ID functions
+  std::size_t& trisegment_ID() const { return mApproximate_traits.mTrisegment_ID ; }
+
+  void reset_trisegment(std::size_t i) const
+  {
+    if ( i+1 == trisegment_ID() )
+    {
+      --trisegment_ID() ;
+      mExact_traits.mTime_cache.Reset(i) ;
+      mApproximate_traits.mTime_cache.Reset(i) ;
+    }
+  }
+
+// functions to initialize (and harmonize) and cache speeds
+  void InitializeLineCoeffs ( CGAL_SS_i::Segment_2_with_ID<K> const& aBorderS )
+  {
+    C2E lToExact ;
+    C2F lToFiltered ;
+
+    mApproximate_traits.InitializeLineCoeffs( lToFiltered(aBorderS) );
+    mExact_traits.InitializeLineCoeffs( lToExact(aBorderS) );
```

Comment:
Could keep a ref to the input halfedge to initialize the exact value (and for harmonize j to i, keep the input halfedge i and not j)

Reply:
Actually that adds a new member to the combinatorial halfedge, and mixes traits and combinatorial items. It will work, but it's a bit nasty. 

I still gained about 30% by caching coeffs, which includes computing all exact coeffs non-lazily. I would rather postpone these lazy exact coeffs till we really do a phase of speed improvements and not robustification.